packages/kbn-apm-synthtrace/src/scenarios/helpers/logs_mock_data.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { generateShortId } from '@kbn/apm-synthtrace-client';
+import { randomInt } from 'crypto';
+
+// Utility function to get a random element from an array
+const getRandomElement = <T>(arr: T[], index?: number): T => {
```

Comment:
That function does something different from what I expected when I read the name:

- If the `index` is given there is nothing random about it.
- If the `index` is out of bounds it suddenly becomes non-deterministic again.

Is there any way to improve that? What do you think about a utility `getAtIndexOrRandom(arr)`, which is defined like this:

```typescript
const getAtIndexOrRandom = <T>(values: T[]) => (index?: number): T => values[index ?? randomInt(values.length)]
```

This could then be used like this:

```typescript
export const getIpAddress = getAtIndexOrRandom(IP_ADDRESSES);
```

What do you think?

Reply:
I think we shouldn't hide the out-of-bounds access. Otherwise the author will never realize it's not working as intended.